src/routing/routing.component.tsx

```diff
@@ -74,28 +75,35 @@ class Routing extends React.Component<
           [this.props.classes.containerShift]: this.props.drawerOpen,
         })}
       >
-        <Switch>
-          <Route exact path="/" component={HomePage} />
-          <Route exact path="/contact" component={ContactPage} />
-          <Route exact path="/help" component={HelpPage} />
-          {this.props.userIsloggedIn && this.props.userIsAdmin ? (
-            <Route exact path="/admin" component={withAuth(AdminPage)} />
-          ) : null}
-          <Route exact path="/login" component={LoginPage} />
-          <Route exact path="/cookies" component={CookiesPage} />
-          {this.props.maintenance.show ? (
-            <Route component={MaintenancePage} />
-          ) : (
-            this.props.plugins.map((p) => (
-              <Route
-                key={`${p.section}_${p.link}`}
-                path={p.link}
-                render={() => <AuthorisedPlugin id={p.plugin} />}
-              />
-            ))
-          )}
-          <Route component={withAuth(PageNotFound)} />
-        </Switch>
+        {/* Redirect to a homepageUrl if set. Otherwise, route to / */}
+        {this.props.homepageUrl &&
+        this.props.location === '/' &&
+        this.props.homepageUrl !== '/' ? (
+          <Redirect to={this.props.homepageUrl} />
+        ) : (
+          <Switch>
+            <Route exact path="/" component={HomePage} />
```

Comment:
Can this be refactored to have one "HomePage" route like this:

```
<Route exact path="/">
  {this.props.homepageUrl && this.props.homepageUrl !== '/'  ? <Redirect to={this.props.homepageUrl} /> : <HomePage/>}
</Route>
```

Reply:
I see what you did there. Much cleaner

Reply:
https://reactrouter.com/web/api/Redirect - our use case is quite similar to the react-router example!